src/oscar/templates/oscar/catalogue/detail.html

```diff
@@ -66,7 +66,7 @@
             {% if user.is_staff %}
                 <a class="pull-right hidden-xs" href="{% url 'dashboard:catalogue-product' pk=product.id %}"><small><i class="icon-edit"></i> {% trans "Edit this product" %}</small></a>
             {% endif %}
-            <h1>{{ product.get_title }}</h1>
+            <h1>{{ product.get_title }} {% if product.is_child and product.attribute_summary %}({{ product.attribute_summary }}){% endif %}</h1>
```

Comment:
As above I don't think attribute summary can be used in titles, because you cannot control its length.